rental/src/main/java/io/axoniq/demo/bikerental/rental/ui/Simulator.java

```diff
@@ -0,0 +1,102 @@
+import org.axonframework.messaging.responsetypes.ResponseTypes;
+import org.axonframework.queryhandling.QueryGateway;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+@Component
+@Profile("simulator")
+@EnableAsync
+public class Simulator {
+
+    private final CommandGateway commandGateway;
+    private final QueryGateway queryGateway;
+    private final BikeRentalDataGenerator bikeRentalDataGenerator;
+
+    @Value("${inventory.size}")
+    private int inventorySize;
+
+    @Value("${inventory.bikeType}")
+    private String inventoryBikeType;
+
+    @Value("${rentalSimulation.bikeType}")
+    private String rentalBikeType;
+
+    @Value("${rentalSimulation.loops}")
+    private int loops;
+
+    @Value("${rentalSimulation.concurrency}")
+    private int concurrency;
+
+    @Value("${rentalSimulation.abandonPaymentFactor}")
+    private int abandonPaymentFactor;
+
+    @Value("${rentalSimulation.delayBetweenLoops}")
+    private int delayBetweenLoops;
+
+    Logger logger = LoggerFactory.getLogger(Simulator.class);
+
+    public Simulator(CommandGateway commandGateway, QueryGateway queryGateway, BikeRentalDataGenerator bikeRentalDataGenerator) {
+        this.commandGateway = commandGateway;
+        this.queryGateway = queryGateway;
+        this.bikeRentalDataGenerator = bikeRentalDataGenerator;
+    }
+
+    public void updateInventoryCreationConfiguration(int sizeOfBikeInventory, String bikeType) {
+        this.inventorySize = sizeOfBikeInventory;
+        this.inventoryBikeType = bikeType;
+    }
+
+    public void updateRentalGenerationConfiguration(String rentalBikeType, int loops, int concurrency, int abandonPaymentFactor, int delay ){
+        this.rentalBikeType = rentalBikeType;
+        this.loops = loops;
+        this.concurrency = concurrency;
+        this.abandonPaymentFactor = abandonPaymentFactor;
+        this.delayBetweenLoops = delay;
+    }
+
+    @Scheduled(fixedRate = 25000, initialDelay = 5000)
+    private void generateData() {
+        try{
+            this.generateBikes();
+        } catch (Exception ex) {
+            logger.error("error generating inventory", ex);
+        }
+
+        this.generateRentals();
+    }
+
+    public void generateBikes() throws ExecutionException, InterruptedException {
+
+        var query = new CountOfBikesByTypeQuery(this.inventoryBikeType);
+        long currentBikeCount = queryGateway.query(query,
+                                                    ResponseTypes.instanceOf(long.class)).get();
+
+        if (currentBikeCount < this.inventorySize) {
+            for (int i = 0; i < 10; i++) {
```

Comment:
This only creates 10 bikes per iteration, not the number established by `inventorySize` (although after `inventorySize/10` iterations it will reach the desired amount of bikes in the system).

Is this the intended behaviour? or do we want to create all the bikes from the beginning and create any missing bikes on each iteration?

I expected this to be `for (int i = 0; i < (inventorySize - currentBikeCount); i++) { ...`

If the idea is to progressively create more bikes, maybe we should refactor/rename the code to explicitly state the intend.

Reply:
I can create them all.  What I was trying to simulate was just bikes being added overtime while rentals were going on.